omc3/correction/response_io.py

```diff
@@ -4,104 +4,134 @@
 
 Input and output functions for response matrices.
 """
+
+from __future__ import annotations
+
+import logging
+import warnings
 from collections import defaultdict
+from collections.abc import Sequence
+from contextlib import contextmanager
 from pathlib import Path
-from typing import Dict, Sequence, Set
 
 import pandas as pd
-import logging
-
+from tables import NaturalNameWarning
 
 LOG = logging.getLogger(__name__)
 
-COMPLIB = 'blosc'  # zlib is the standard compression
-COMPLEVEL = 9  # goes from 0-9, 9 is highest compression, None deactivates if COMPLIB is None
+COMPLIB: str = "blosc"  # zlib is the standard compression
+COMPLEVEL: int = 9  # goes from 0-9, 9 is highest compression, None deactivates if COMPLIB is None
 
 
 # Fullresponse -----------------------------------------------------------------
 
 
-def read_fullresponse(path: Path, optics_parameters: Sequence[str] = None) -> Dict[str, pd.DataFrame]:
+def read_fullresponse(path: Path, optics_parameters: Sequence[str] = None) -> dict[str, pd.DataFrame]:
     """Load the response matrices from disk.
     Beware: As empty DataFrames are skipped on write,
     default for not found entries are empty DataFrames.
     """
     if not path.exists():
         raise IOError(f"Fullresponse file {str(path)} does not exist.")
-
     LOG.info(f"Loading response matrices from file '{str(path)}'")
-    with pd.HDFStore(path, mode='r') as store:
-        _check_keys(store, optics_parameters, 'fullresponse')
-
-        fullresponse = defaultdict(pd.DataFrame)
-        if optics_parameters is None:
-            optics_parameters = _main_store_groups(store)
-        for param in optics_parameters:
-            fullresponse[param] = store[param]
+
+    # If encountering issues, remove the context manager and debug
+    with ignore_natural_name_warning():
+        with pd.HDFStore(path, mode="r") as store:
+            _check_keys(store, optics_parameters, "fullresponse")
+
+            fullresponse = defaultdict(pd.DataFrame)
+            if optics_parameters is None:
+                optics_parameters = _main_store_groups(store)
+            for param in optics_parameters:
+                fullresponse[param] = store[param]
```

Comment:
This addresses `NaturalNameWarning`s happening during HDF5 tables operations. This is because we have keys such as `kq4.r8b2` which pytables wants to raise awareness about (we can't do `table.kq4.r8b2` as this resolves as two attributes accesses).

It is not avoidable due to our variables, and not a problem since `pandas` properly handles access operations for us (via `getattr`).

Reply:
Same for other places in this file which call this context manager.